File: ruby/EnemyStarShip.rb
#encoding-utf: 8

require_relative 'EnemyToUI'
require_relative 'ShotResult'

module Deepspace
    class EnemyStarShip
        def initialize(n, a, s, l, d)   #string, float, float, loot, damage
            @name = n
            @ammoPower = a
            @shieldPower = s
            @loot = l
            @damage = d
        end

        def self.newCopy(e)
            EnemyStarShip.new(e.name, e.ammoPower, e.shieldPower, e.loot, e.damage)
        end

        def getUIversion
            return EnemyToUI.new(self)
        end

        attr_reader :name
        attr_reader :ammoPower
        attr_reader :shieldPower
        attr_reader :loot
        attr_reader :damage

        def protection
            @shieldPower
        end

        def fire
            @ammoPower
        end

        def receiveShot(shot)
            if protection < shot
                return ShotResult::DONOTRESIST
            else return ShotResult::RESIST
            end
        end

        def to_s
            l = "vacio"
            if loot != nil
                l = @loot.to_s
            end

            d = "vacio"
            if damage != nil
                d = @damage.to_s
            end

            return "Name: #{@name}, AmmoPower: #{@ammoPower}, ShieldPower: #{@shieldPower}, Loot: " + l + ", Damage: " + d
        end
    end
end

File: java/Deepspace/src/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author alvaro
 */
public class EnemyStarShip {
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Loot loot;
    private Damage damage;
    
    EnemyStarShip(String n, float a, float s, Loot l, Damage d){
        name = n;
        ammoPower = a;
        shieldPower = s;
        loot = l;
        damage = d;
    }
    
    EnemyStarShip(EnemyStarShip e){
        name = e.getName();
        ammoPower = e.getAmmoPower();
        shieldPower = e.getShieldPower();
        loot = e.getLoot();
        damage = e.getDamage(); 
    }
    
    EnemyToUI getUIversion(){
        return new EnemyToUI(this);
    }
    
    public float fire(){
        return ammoPower;
    }
    
    public float getAmmoPower(){
        return ammoPower;
    }
    
    public Damage getDamage(){
        return damage;
    }
    
    public Loot getLoot(){
        return loot;
    }
    
    public String getName(){
        return name;
    }
    
    public float getShieldPower(){
        return shieldPower;
    }
    
    public float protection() {
        return shieldPower;
    }
    
    public ShotResult receiveShot (float shot) {
        if(shot > shieldPower)
            return ShotResult.DONOTRESIST;
        else
            return ShotResult.RESIST;
    }
    
    @Override
    public String toString(){
        String result = "por hacer";
        
        String n = "Nombre: " + this.name + "\n";
        String sp = "ShieldPower: " + this.shieldPower + "\n";
        String ap = "AmmoPower: " + this.ammoPower + "\n";
        String l = "Loot: " + loot.toString();
        String d = "Damage: " + damage.toString();
                
        result = n + sp + ap + l + d;
        
        return result;
    }
    
}
